packages/oauth/oauth-provider/src/oauth-provider.ts

```diff
@@ -1145,28 +1144,27 @@ export class OAuthProvider extends OAuthVerifier {
      * Wrap an OAuth endpoint in a middleware that will set the appropriate
      * response headers and format the response as JSON.
      */
-    const jsonHandler = <T, TReq extends Req, TRes extends Res, Json>(
-      buildJson: (this: T, req: TReq, res: TRes) => Awaitable<Json>,
-      status?: number,
+    const jsonHandler = <T, TReq extends Req, TRes extends Res, Payload>(
+      buildJson: (
+        this: T,
+        req: TReq,
+        res: TRes,
+      ) => Awaitable<{ payload: Payload; status?: number }>,
     ): Handler<T, TReq, TRes> =>
       async function (req, res) {
-        try {
-          // https://www.rfc-editor.org/rfc/rfc6749.html#section-5.1
-          res.setHeader('Cache-Control', 'no-store')
-          res.setHeader('Pragma', 'no-cache')
-
-          // Ensure we can agree on a content encoding & type before starting to
-          // build the JSON response.
-          if (!negotiateContent(req, ['application/json'])) {
-            throw createHttpError(406, 'Unsupported media type')
-          }
+        // https://www.rfc-editor.org/rfc/rfc6749.html#section-5.1
+        res.setHeader('Cache-Control', 'no-store')
+        res.setHeader('Pragma', 'no-cache')
+
+        // Ensure we can agree on a content encoding & type before starting to
+        // build the JSON response.
+        if (!negotiateContent(req, ['application/json'])) {
+          return res.writeHead(406).end()
+        }
```

Comment:
Unrelated fix. Because the code here requires the caller to "accept" json, we make sure to always return json. It makes no sense to require a called to accept json if we are going to answer 204 "no-content".

This also makes all API responses more consistent